components/datetime/src/any/zoned_datetime.rs

```diff
@@ -338,10 +338,10 @@ impl ZonedDateTimeFormatter {
     ///     length::Date::Medium,
     ///     length::Time::Long,
     /// );
-    /// let locale = locale!("en-u-ca-gregory");
+    /// let locale = locale!("en-u-ca-gregory").into();
     ///
     /// let zdtf = ZonedDateTimeFormatter::try_new(
-    ///     &locale.into(),
+    ///     &locale,
```

Comment:
Note: All of these examples assume that Preferences will be borrowed. Have we had that discussion? Because it could also be a Copy type that borrows the LanguageIdentifier.